src/App/App.test.js

```diff
@@ -1,8 +1,114 @@
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import React from "react";
+import App from "./index";
+import { render, screen, waitFor, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import "@testing-library/jest-dom";
+import { Router, MemoryRouter } from "react-router";
+import { createMemoryHistory } from "history";
+import { addToMySet } from "../actions";
+import { fetchTenJokes } from "../apiCalls";
+import mockData from "../mockData";
+import { act } from "react-dom/test-utils";
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+jest.mock("../apiCalls");
+jest.mock("../actions");
+
+describe("App", () => {
+  beforeEach(() => {
+    fetchTenJokes.mockResolvedValueOnce(mockData.jokes);
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+  });
+  it("should render the NavBar on load with DevWit App name", () => {
+    const nav = screen.getByRole("navigation");
+    expect(nav).toBeInTheDocument();
+  });
+
+  it("should redirect URL path to /home on load", () => {
+    const history = createMemoryHistory();
+
+    render(
+      <Router history={history}>
+        <App />
+      </Router>
+    );
+    expect(history.location.pathname).toBe("/home");
+  });
+
+  it("should render Home component on load", () => {
+    const welcomeMsg = screen.getByRole("heading", { name: /welcome, dev\./i });
```

Comment:
Testing Playground is soooo useful! Right?!

Reply:
YES! Thank you so much for tipping me off on this resource!